tools/src/test/java/org/apache/kafka/tools/LogDirsCommandTest.java

```diff
@@ -16,33 +16,153 @@
+            int brokerId = 0;
+            String output = assertDoesNotThrow(() -> execute(fromArgsToOptions("--bootstrap-server", clusterInstance.bootstrapServers(), "--broker-list", String.valueOf(brokerId), "--describe"), admin));
+
+            // check only broker in list is present
+            assertTrue(output.contains("\"broker\":" + brokerId));
+            clusterInstance.brokerIds().stream().filter(id -> id != brokerId).forEach(id -> {
+                assertFalse(output.contains("\"broker\":" + id));
+            });
+
+            // check log dir and topic partition are present
+            Map<Integer, Map<String, LogDirDescription>> logDirs = assertDoesNotThrow(() -> admin.describeLogDirs(Collections.singleton(brokerId)).allDescriptions().get());
+            assertEquals(1, logDirs.size());
+            logDirs.forEach((brokerIdValue, logDirInfo) -> {
+                assertFalse(logDirInfo.isEmpty());
+                logDirInfo.forEach((logDir, logDirInfoValue) -> {
+                    assertTrue(output.contains("\"logDir\":\"" + logDir + "\""));
+                    Optional<TopicPartition> topicPartition = logDirInfoValue.replicaInfos().keySet().stream().filter(tp -> tp.topic().equals(TOPIC)).findFirst();
+                    assertTrue(topicPartition.isPresent());
+                    assertTrue(output.contains("\"partition\":\"" + topicPartition.get() + "\""));
+                });
+            });
+        }
+    }
+
+    @ClusterTest
+    public void testLogDirsWithNonExistentTopic(ClusterInstance clusterInstance) {
+        try (Admin admin = Admin.create(Collections.singletonMap(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, clusterInstance.bootstrapServers()))) {
+            String output = assertDoesNotThrow(() -> execute(fromArgsToOptions("--bootstrap-server", clusterInstance.bootstrapServers(), "--topic-list", TOPIC, "--describe"), admin));
+            // check all brokers are present
+            clusterInstance.brokerIds().forEach(brokerId -> {
+                assertTrue(output.contains("\"broker\":" + brokerId));
+            });
+
+            // check log dir is present, but topic partition is not
+            Map<Integer, Map<String, LogDirDescription>> logDirs = assertDoesNotThrow(() -> admin.describeLogDirs(clusterInstance.brokerIds()).allDescriptions().get());
+            assertFalse(logDirs.isEmpty());
+            logDirs.forEach((brokerId, logDirInfo) -> {
+                logDirInfo.forEach((logDir, logDirInfoValue) -> {
+                    assertTrue(output.contains("\"logDir\":\"" + logDir + "\""));
+                    logDirInfoValue.replicaInfos().forEach((topicPartition, replicaInfo) -> {
+                        assertFalse(output.contains("\"partition\":\"" + topicPartition + "\""));
+                    });
+                });
+            });
+        }
+    }
+
+    @ClusterTest
+    public void testLogDirsWithSpecificTopic(ClusterInstance clusterInstance) {
```

Comment:
Could you please add test to verify that it returns only the specific topic will get returned? for example, you can create another topic before testing.